### PR TITLE
actions on cluster tests

### PR DESCRIPTION
**Pull request by yonasberhe23**
### Summary
This PR covers the functionality in the Cluster actions (within ellipsis)
View Test Plan https://github.com/rancher/qa-tasks/issues/753

### Technical notes summary
Between the two download tests, the kubeconfig and YAML files are created with the same name in the downloads directory which is causing failures. The easiest solution was to add the [cypress-delete-downloads-folder ](https://www.npmjs.com/package/cypress-delete-downloads-folder) npm package to delete the downloads directory to avoid the conflicting files.
Changes include adding several lines of code to our Cypress configuration files which allows the cleanup of our downloads with `cy.deleteDownloadsFolder()` command.

### Screenshots
Screenshot 2023-10-18 at 1 17 16 PM

### CODE REVIEW

File: cypress/e2e/tests/pages/cluster-manager.spec.ts [hunk 1]
```diff
@@ -115,6 +134,25 @@ describe('Cluster Manager', { tags: '@adminUser' }, () => {
         });
       });
 
+      it('can download YAML', () => {
+        // Delete downloads directory. Need a fresh start to avoid conflicting file names
+        cy.deleteDownloadsFolder();
```
Comment (richard-cox) [inline cypress/e2e/tests/pages/cluster-manager.spec.ts:139]: i'm being paranoid... but to confirm, this just deletes whatever download folder is associated with an isolated cypress browser session?

File: cypress/e2e/tests/pages/cluster-manager.spec.ts [hunk 2]
```diff
@@ -280,4 +318,10 @@ describe('Cluster Manager', { tags: '@adminUser' }, () => {
 
     clusterDashboard.waitForPage(undefined, 'cluster-events');
   });
+
+  it('can connect to kubectl shell', () => {
+    clusterList.goTo();
+    clusterList.list().actionMenu('local').getMenuItem('Kubectl Shell').click();
+    cy.contains('Connected').should('be.visible');
```
Comment (richard-cox) [inline cypress/e2e/tests/pages/cluster-manager.spec.ts:325]: This should use `shell.po`

File: cypress/e2e/tests/pages/cluster-manager.spec.ts [hunk 3]
```diff
@@ -72,6 +72,25 @@ describe('Cluster Manager', { tags: '@adminUser' }, () => {
         detailRKE2ClusterPage.waitForPage(undefined, 'registration');
       });
 
+      it('can copy config to clipboard', () => {
+        clusterList.goTo();
```
Comment (richard-cox) [inline cypress/e2e/tests/pages/cluster-manager.spec.ts:76]: can these three new tests use navTo?

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: cypress/e2e/tests/pages/cluster-manager.spec.ts (hunk 1 above)
Comment (yonasberhe23) [reply cypress/e2e/tests/pages/cluster-manager.spec.ts:139]: That is correct.

File: cypress/e2e/tests/pages/cluster-manager.spec.ts [hunk 4]
```diff
@@ -72,6 +73,25 @@ describe('Cluster Manager', { tags: '@adminUser' }, () => {
...
 
+      it('can copy config to clipboard', () => {
+        ClusterManagerListPagePo.navTo();
+
+        cy.intercept('POST', '*action=generateKubeconfig').as('copyKubeConfig');
+        clusterList.list().actionMenu(rke2CustomName).getMenuItem('Copy KubeConfig to Clipboard').click();
+        cy.wait('@copyKubeConfig');
+
+        // Verify confirmation message displays and is hidden after ~3 sec
+        cy.get('.growl-text').contains('Copied KubeConfig to Clipboard').should('be.visible');
```
Comment (richard-cox) [inline cypress/e2e/tests/pages/cluster-manager.spec.ts:84]: Comment only.

At some point we should make a growl PO. If we use it more we should do so, but at the moment this is fine.

**Review by richard-cox — APPROVED** [review]

**Pull request merged**